shell/browser/web_contents_preferences.cc

```diff
@@ -159,7 +159,7 @@ void WebContentsPreferences::Clear() {
   safe_dialogs_ = false;
   safe_dialogs_message_ = absl::nullopt;
   ignore_menu_shortcuts_ = false;
-  background_color_ = absl::nullopt;
+  background_color_ = SK_ColorTRANSPARENT;
```

Comment:
I'm not sure if this is correct 🤔 I think this should only be `SK_ColorTRANSPARENT` **if** the owning BrowserWindow is transparent 🤔 I think WebContentsPreferences _used_ to have a concept of transparency but apparently not any more

Reply:
That was the intention with using the optional value.

The `backgroundColor` web preferences property was changed to use a hidden field. I wonder if this also needs to be the same with BrowserViews somewhere?

https://github.com/electron/electron/blob/c5b517d89f8c156caa7d33297bd43dcdbaf8d063/shell/browser/api/electron_api_browser_window.cc#L40-L50

Also, just a heads up that the v14 implementation differs a bit because of the WebPreferences refactor. Backport might be painful 😞

Reply:
btw, I have some wip tests written for backgroundColor so hopefully we'll be able to catch these issues in the future! 🤞 https://github.com/electron/electron/compare/main...samuelmaddock:test/bw-bg-color?expand=1

Reply:
A little more investigation: BrowserView might not be calling `WebContents::SetPageBaseBackgroundColor()`? 🤔

https://github.com/electron/electron/blob/b491a4c82f608e8bbbc0e08d3da1ba6897fbee2b/shell/browser/api/electron_api_browser_view.cc#L149-L151

Reply:
Yeah, I think you're right - it may not be calling it or there may be a painting/timing error on when it's called?

I actually found this bug yesterday while digging into this more: #30993 If I start a transparent window in Fiddle, the background renders as white, but then sometimes will switch back to transparent if I alt+tab away. Maybe there's a timing issue where background_color for BrowserView sometimes isn't being set before the first paint? Looking into it more today! 🙂

Reply:
@samuelmaddock You were right - it looks like there's an issue with `SetPageBaseBackgroundColor`, where transparent windows would be white until a repaint (e.g. resize) happened. It looks like Chrome's going to merge the two APIs and address this down the line (https://chromium-review.googlesource.com/c/chromium/src/+/2958369), but we can get around it now by setting the background directly on the render widget host view. There's a new PR open to address this (linked below) 🙂